### ISSUE TITLE
Javascript type inference problem for classes if constructor argument number doesn't match

### ISSUE THREAD

**Issue by mjbvz** [Microsoft/TypeScript#13032]
_From @chanep on December 19, 2016 13:53_

- VSCode Version: 1.8.0
- OS Version: Any

VScode doesn't detect correctly instance type if class constructor is called with different number of arguments. 

Example:

```js
class Person{
    constructor(arg){
        this.arg = arg;
    }
}

let p1 = new Person(null);      //OK, p1 is type Person
let p2 = new Person();           //WRONG, p2 is type "any"
```

I think it is a little annoying to have to put null's all over the place

_Copied from original issue: Microsoft/vscode#17539_

**Comment by mjbvz** [comment]
For JavaScript, I think that TypeScript should be more relaxed when looking at constructor arguments.  Users can currently workaround this issue by explicitly specifying that `arg` is optional:

```js
class Person{
    constructor(arg = null){
        this.arg = arg;
    }
}
```

but leaving arguments implicitly specified is still a common pattern.

**Comment by RyanCavanaugh** [comment]
I believe we do this correctly for call signatures, but it's not working for construct signatures for some reason

**Comment by chanep** [comment]
Hi, is there any news abnout this?

**Comment by RyanCavanaugh** [comment]
No longer repros - seems this has been fixed
